executor/inspection_result.go

```diff
@@ -287,9 +291,102 @@ func (configInspection) inspectCheckConfig(_ context.Context, sctx sessionctx.Co
 			})
 		}
 	}
+	results = append(results, c.checkTiKVBlockCacheSizeConfig(ctx, sctx, filter)...)
 	return results
 }
 
+func (c configInspection) checkTiKVBlockCacheSizeConfig(ctx context.Context, sctx sessionctx.Context, filter inspectionFilter) []inspectionResult {
+	item := "storage.block-cache.capacity"
+	if !filter.enable(item) {
+		return nil
+	}
+	sql := "select instance,value from information_schema.cluster_config where type='tikv' and `key` = 'storage.block-cache.capacity'"
+	rows, _, err := sctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQLWithContext(ctx, sql)
+	if err != nil {
+		sctx.GetSessionVars().StmtCtx.AppendWarning(fmt.Errorf("check configuration in reason failed: %v", err))
+	}
+	extractIP := func(addr string) string {
+		if idx := strings.Index(addr, ":"); idx > -1 {
+			return addr[0:idx]
+		}
+		return addr
+	}
+
+	ipToBlockSize := make(map[string]uint64)
+	ipToCount := make(map[string]int)
+	for _, row := range rows {
+		ip := extractIP(row.GetString(0))
+		size, err := c.convertReadableSizeToByteSize(row.GetString(1))
+		if err != nil {
+			sctx.GetSessionVars().StmtCtx.AppendWarning(fmt.Errorf("check TiKV block-cache configuration in reason failed: %v", err))
+			return nil
+		}
+		ipToBlockSize[ip] += size
+		ipToCount[ip]++
+	}
+
+	sql = "select instance, value from metrics_schema.node_total_memory where time=now()"
+	rows, _, err = sctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQLWithContext(ctx, sql)
+	if err != nil {
+		sctx.GetSessionVars().StmtCtx.AppendWarning(fmt.Errorf("check configuration in reason failed: %v", err))
+	}
+	ipToMemorySize := make(map[string]float64)
+	for _, row := range rows {
+		ip := extractIP(row.GetString(0))
+		size := row.GetFloat64(1)
+		ipToMemorySize[ip] += size
+	}
+
+	var results []inspectionResult
+	for ip, blockSize := range ipToBlockSize {
+		if memorySize, ok := ipToMemorySize[ip]; ok {
+			if float64(blockSize) > memorySize*0.45 {
+				detail := fmt.Sprintf("There are %v TiKV server in %v node, the total 'storage.block-cache.capacity' of TiKV is more than (0.45 * total node memory)",
+					ipToCount[ip], ip)
+				results = append(results, inspectionResult{
+					tp:       "tikv",
+					instance: ip,
+					item:     item,
+					actual:   fmt.Sprintf("%v", blockSize),
+					expected: fmt.Sprintf("< %.0f", memorySize*0.45),
+					severity: "warning",
+					detail:   detail,
+				})
+			}
+		}
+	}
+	return results
+}
+
+func (configInspection) convertReadableSizeToByteSize(sizeStr string) (uint64, error) {
+	const KB = uint64(1024)
```

Comment:
Using iota generate them seems better.
const (
    _ = iota
    KB = 1 << (10 * iota) // 1 << (10 * 1)
    MB                            // 1 << (10 * 2)
    GB                            // 1 << (10 * 3)
    TB                            // 1 << (10 * 4)
    PB                            // 1 << (10 * 5)
)

Reply:
em... I still prefer the original code style...